src/software/simulated_tests/simulated_test_fixture.py

```diff
@@ -171,6 +170,14 @@ def runner(
                     world = self.world_buffer.get(
                         block=True, timeout=WORLD_BUFFER_TIMEOUT, return_cached=False
                     )
+                    # AI is usually slower than getting worlds, so that's the limiting factor
+                    # We get a new primitive set and only update the time if one exists to make time dependent on AI
+                    # Otherwise, we end up with behaviour that doesn't simulate what would happen in the real world
+                    primitive_set = self.primitive_set_buffer.get(
+                        block=False, timeout=WORLD_BUFFER_TIMEOUT, return_cached=False
+                    )
```

Comment:
It seems like we now would be ticking the simulation (`SimulatorTick` above) constantly, but only ticking the time here whenever there's a primitive, which might not be after every iteration. i.e. the time the test think has passed is not in sync with the time that's actually passed in sim. 

I feel like the solution would be to make the primitive `get` call also blocking, so we guarantee that for every simulation tick, we receive a World and a Primitive (i.e. fullsystem has fully processed the simulation step and is now waiting on the next tick). 

Since we'll also blocking for longer, it's probably a good idea to update the `time.sleep` above (_which is used for running the test in realtime when we run it with Thunderscope GUI (`-t`), rather than running the test as fast as Worlds are outputted when we run it without a GUI - @suchirss hoping this answers your question_) such that it also take into account how long the blocking call and the rest of the python code takes. For example, if we want to step the simulation every 16ms (~60hz simulation) and the blocking calls + rest of the code here takes 10ms, we want to only sleep for 6ms in order for the visualizer to appear in realtime.